src/g.cpp

```diff
@@ -19,9 +19,9 @@ g1::g1(const g1& e) : x(e.x), y(e.y), z(e.z)
 
 optional<g1> g1::fromJacobianBytesBE(const span<const uint8_t, 144> in, const bool check, const bool raw)
 {
-    optional<fp> x = fp::fromBytesBE(span<const uint8_t, 48>(&in[ 0], &in[ 48]), check, raw);
-    optional<fp> y = fp::fromBytesBE(span<const uint8_t, 48>(&in[48], &in[ 96]), check, raw);
-    optional<fp> z = fp::fromBytesBE(span<const uint8_t, 48>(&in[96], &in[144]), check, raw);
+    optional<fp> x = fp::fromBytesBE(span<const uint8_t, 48>(&in[ 0], &in[ 48]), true, raw);
+    optional<fp> y = fp::fromBytesBE(span<const uint8_t, 48>(&in[48], &in[ 96]), true, raw);
+    optional<fp> z = fp::fromBytesBE(span<const uint8_t, 48>(&in[96], &in[144]), true, raw);
     if(!x.has_value() || !y.has_value() || !z.has_value()) return {};
     g1 p = g1({x.value(), y.value(), z.value()});
```

Comment:
That is a non-idiomatic (and verbose) way to use std::optional. I would prefer:
```suggestion
    if(!x || !y || !z) return {};
    g1 p({*x, *y, *z});
```

Reply:
fixed all similar issues.